src/prometheus/definition.go

```diff
@@ -701,3 +724,42 @@ func getRawEntityID(parentGroupLabel, groupLabel, entityID string, groups defini
 	}
 	return rawEntityID, nil
 }
+
+func FromLabelValueWithLabelInName(rawMetricName, label, format, valueLabel string) definition.FetchFunc {
```

Comment:
I think it would be nice some description here perhaps with an example :)

Reply:
Right. My idea was that when we roughly agree on the implementation, I'll add proper unit tests :)

Reply:
I would rename `label` parameter to something that express that the value of that label will be used in the metric name resulted.
Perhaps `nameLabel` ? but perhaps is not needed with the descriptions and example

Reply:
I agree, that's a good idea.

Reply:
I hate so much fact that this function is doing 2 things and I cannot really easily compose those 2 things together instead of hardcoding them in a single function :smile:

Reply:
I think the following will be more accurate:
```suggestion
// FromLabelValueWithLabelInName returns a FetchFunc that returns metrics with a templatizable name based on a label,
// and whose value is gathered from another label.
//
// `format` will be evaluated using the value of the label `label` and result will be used as metric name.
//
// The value of the metric will be set to the value of `valueLabel` label.
//
// For example, for the following metric:
//
// rawMetricName{foo="whatever",bar="false"} 1
//
// called as:
//
// FromLabelValueWithLabelInName("rawMetricName", "foo", "myMetric.%s", "bar")
//
// the function will produce the following metric:
//
// myMetric.whatever = "false"
//
func FromLabelValueWithLabelInName(rawMetricName, label, format, valueLabel string) definition.FetchFunc {
```

Reply:
Looks great!